src/liftopia/Manipulator/FillCanvasSize.php

```diff
@@ -0,0 +1,75 @@
+<?php
+namespace Liftopia\Manipulator;
+
+use Intervention\Image\Image;
+use Intervention\Image\ImageManagerStatic;
+use League\Glide\Api\Manipulator\Size;
+use Symfony\Component\HttpFoundation\Request;
+
+class FillCanvasSize extends Size
+{
+    /**
+     * Perform size image manipulation.
+     * @param  Request $request The request object.
+     * @param  Image   $image   The source image.
+     * @return Image   The manipulated image.
+     */
+    public function run(Request $request, Image $image)
+    {
+        $width = $this->getWidth($request->get('w'));
+        $height = $this->getHeight($request->get('h'));
+        $fit = $this->getFit($request->get('fit'));
+        $crop = $this->getCrop($request->get('crop'));
+        $bgColor = $this->getBgColor($request->get('bg'));
+
+        list($width, $height) = $this->resolveMissingDimensions($image, $width, $height);
+        list($width, $height) = $this->limitImageSize($width, $height);
+
+        if (round($width) !== round($image->width()) or
```

Comment:
We use `||`. But I copypastad this from the Glide Size manipulator so I could hack in the $bgColor thing.